core/server/master/src/main/java/alluxio/master/file/activesync/ActiveSyncer.java

```diff
@@ -61,11 +68,16 @@ public ActiveSyncer(FileSystemMaster fileSystemMaster, ActiveSyncManager syncMan
     mSyncManager = syncManager;
     mMountId = mountId;
     mMountTable = mountTable;
+    mSyncTasks = new LinkedBlockingQueue<>(32);
   }
 
   @Override
   public void heartbeat() {
     LOG.debug("start Active Syncer heartbeat");
+    // Remove any previously completed sync tasks
+    while (mSyncTasks.peek() != null && !mSyncTasks.peek().isDone()) {
```

Comment:
should this be `mSyncTasks.peek().isDone()`? Don't we want to remove the ones that are considered done?